src/utilities/theme/utils.ts

```diff
@@ -1,115 +1,304 @@
 import {isLight} from '../color-validation';
-import {constructColorName} from '../color-names';
-import {createLightColor} from '../color-manipulation';
-import {compose} from '../compose';
-
-import {Theme, ColorsToParse, ThemeVariant, ThemeColors} from './types';
-import {needsVariantList} from './config';
-
-export function setColors(theme: Theme | undefined): string[][] | undefined {
-  let colorPairs;
-  if (theme && theme.colors) {
-    Object.entries(theme.colors).forEach(([colorKey, pairs]) => {
-      const colorKeys = Object.keys(pairs);
-      if (colorKey === 'topBar' && colorKeys.length > 1) {
-        colorPairs = colorKeys.map((key: string) => {
-          const colors = (theme.colors as ThemeColors).topBar;
-          return [constructColorName(colorKey, key), colors[key]];
-        });
-      } else {
-        colorPairs = parseColors([colorKey, pairs]);
-      }
-    });
-  }
-
-  return colorPairs;
-}
-
-export function needsVariant(name: string) {
-  return needsVariantList.indexOf(name) !== -1;
-}
-
-const lightenToString: (
-  color: HSLColor | string,
+import {ThemeConfig, Theme, CustomPropertiesLike} from './types';
+import {setColors} from './legacy-utils';
+
+// TODO: Move this to a settings prop on app provider and
+// make available through context
+enum FakeSetting {
+  Theming = 1,
+  // Theming = 0,
+}
+
+export function buildCustomProperties(
+  themeConfig: ThemeConfig,
+): CustomPropertiesLike {
+  return {
+    ...setColors(themeConfig),
+    ...Colors(themeConfig),
+  };
+}
+
+export function buildThemeContext(themeConfig: ThemeConfig): Theme {
+  const {logo} = themeConfig;
+  return {logo};
+}
+
+function Colors(theme: ThemeConfig) {
+  const {colors = {}} = theme;
+
+  if (FakeSetting.Theming === 0) return;
+
+  const {
+    // surface = '#111213',
+    surface = '#FAFAFA',
+    onSurface = '#1F2225',
+    interactive = '#0870D9',
+    interactiveNeutral = '#EAEAEB',
+    branded = '#008060',
+    critical = '#E32727',
+    warning = '#FFC453',
+    highlight = '#59D0C2',
+    success = '#008060',
+  } = colors;
+
+  const lightSurface = isLight(hslToRgb(colorToHsla(surface)));
+
+  return {
+    ...customPropertyTransformer({
+      ...surfaceColors(colorToHsla(surface), lightSurface),
+      ...onSurfaceColors(colorToHsla(onSurface), lightSurface),
+      ...interactiveColors(colorToHsla(interactive), lightSurface),
+      ...interactiveNeutralColors(
+        colorToHsla(interactiveNeutral),
+        lightSurface,
+      ),
+      ...brandedColors(colorToHsla(branded), lightSurface),
+      ...criticalColors(colorToHsla(critical), lightSurface),
+      ...warningColors(colorToHsla(warning), lightSurface),
+      ...highlightColors(colorToHsla(highlight), lightSurface),
+      ...successColors(colorToHsla(success), lightSurface),
+    }),
+    ...overrides(),
+  };
+}
+
+function surfaceColors(
+  color: HSLAColor,
+  lightSurface: boolean,
+): {
+  [key: string]: HSLAColor;
```

Comment:
Don't specify a return type here, you're making the type be a lot wider than it needs to be - this objec has a specific set of keys and it'd be useful to identify that shape rather than be "and object that contain any key".